src/materials/Material.js

```diff
@@ -506,6 +506,18 @@ class Material extends EventDispatcher {
 
 	dispose() {
 
+		for ( const i in this ) {
+
+			const value = this[ i ];
+
+			if ( value !== null ) {
+
+				if ( value.isTexture ) value.dispose();
```

Comment:
You are making assumptions in this PR which are not correct. E.g. textures can be shared in materials, shader so it is no correct to call `dispose()` on them if the material gets disposed of.